### PR TITLE
Migrate functions used in ExpressionEditorTextfield to MLv2 query

### PR DESCRIPTION
**Pull request by uladzimirdev**
closes https://github.com/metabase/metabase/issues/36235
closes https://github.com/metabase/metabase/issues/36406
closes https://github.com/metabase/metabase/issues/36254

This PR includes migrating to MLv2 query of the next functions:

- `diagnose`
- `format`
- `processSource`

and all the related files and functions which work with Expressions.

### CODE REVIEW

File: frontend/src/metabase-lib/expressions/index.ts [hunk 1]
```diff
@@ -116,29 +146,55 @@ export function parseDimension(
...
+    .find(column => {
+      const displayInfo = Lib.displayInfo(query, stageIndex, column);
+
+      return EDITOR_FK_SYMBOLS.symbols.some(separator => {
+        const displayName = getDisplayNameWithSeparator(
+          displayInfo.longDisplayName,
+          separator,
+        );
+
+        return displayName === name;
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/expressions/index.ts:179]: Here

File: frontend/src/metabase-lib/expressions/index.ts [hunk 2]
```diff
@@ -300,19 +338,13 @@ export function isDimension(expr: unknown): boolean {
 
 export function isMetric(expr: unknown): boolean {
   return (
-    Array.isArray(expr) &&
-    expr[0] === "metric" &&
-    expr.length === 2 &&
-    typeof expr[1] === "number"
+    Array.isArray(expr) && expr[0] === "metric" && typeof expr[1] === "number"
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/expressions/index.ts:341]: MLv2 returns 3 parameters

Comment (ranquild) [reply frontend/src/metabase-lib/expressions/index.ts:341]: Since you remove last options in `legacyFieldRef`, do we need this?

**Review by ranquild — COMMENTED** [review]
One question

File: frontend/src/metabase-lib/expressions/index.ts (hunk 2 above)
Comment (uladzimirdev) [reply frontend/src/metabase-lib/expressions/index.ts:341]: I will revert it back, but the check looks weird to me to be honest

Comment (uladzimirdev) [reply frontend/src/metabase-lib/expressions/index.ts]: https://github.com/metabase/metabase/pull/36568/commits/a5d0076d84e40bff6878ac5bc1e1c536347bf10f

**Review by ranquild — APPROVED** [review]

**Pull request merged**